pkg/webhook/certs.go

```diff
@@ -44,9 +44,10 @@ var crLog = logf.Log.WithName("cert-rotation")
 
 var vwhGVK = schema.GroupVersionKind{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "ValidatingWebhookConfiguration"}
 
-var _ manager.Runnable = &certRotator{}
+var _ manager.Runnable = &CertRotator{}
 
-func AddRotator(mgr manager.Manager, cr *certRotator, vwhName string) error {
+// AddRotator adds the CertRotator and ReconcileVWH to the manager.
+func AddRotator(mgr manager.Manager, cr *CertRotator, vwhName string) error {
 	// Use a new client so we are unaffected by the cache sync kill signal
```

Comment:
We no longer need to create a second client if we are ensuring that rotation happens before other controllers are added, as we no longer risk a premature tear-down. We should remove this code to save on memory.

Reply:
SG, will remove it.

Is it that this new client won't be affected when the manager stops? So that this client can still write secret in the previous 5-s sleep?

Reply:
The manager usually shuts down the client before it exits, which means the 5 second sleep would have never helped. By creating a second client, we bypassed the manager's shut down logic at the cost of duplicating the cache a bit. Now that the manager is no longer tearing down, this should no longer be necessary.